src/Symfony/Component/HttpClient/EventSourceHttpClient.php

```diff
@@ -0,0 +1,151 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpClient;
+
+use Symfony\Component\HttpClient\Chunk\ServerSentEvent;
+use Symfony\Component\HttpClient\Exception\EventSourceException;
+use Symfony\Component\HttpClient\Response\AsyncContext;
+use Symfony\Component\HttpClient\Response\AsyncResponse;
+use Symfony\Contracts\HttpClient\ChunkInterface;
+use Symfony\Contracts\HttpClient\Exception\TransportExceptionInterface;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+use Symfony\Contracts\HttpClient\ResponseInterface;
+
+/**
+ * @author Antoine Bluchet <soyuka@gmail.com>
+ * @author Nicolas Grekas <p@tchwork.com>
+ */
+final class EventSourceHttpClient implements HttpClientInterface
+{
+    use AsyncDecoratorTrait;
+    use HttpClientTrait;
+
+    private $reconnectionTime;
+
+    public function __construct(HttpClientInterface $client = null, float $reconnectionTime = 2.0)
+    {
+        $this->client = $client ?: HttpClient::create();
+        $this->reconnectionTime = $reconnectionTime;
+    }
+
+    public function connect($method, string $url, array $options = []): ResponseInterface
```

Comment:
Would it make sense to have an interface for this one?
I'm not sure, just wondering.

Reply:
I don't think so but let's gather opinions.

Reply:
I'm also in favor of having a dedicated interface. The user may want to provide its own EventSource implementation (for instance, to support a server not respecting strictly the standard or something like that).

Reply:
I think I'd prefer having the new interface in the component, not in the contracts.